crates/bevy_ecs/src/identifier/kinds.rs

```diff
@@ -0,0 +1,11 @@
+/// The kinds of ID that [`super::Identifier`] can represent. Each
+/// variant imposes different usages of the low/high segments
+/// of the ID.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(u8)]
+pub(crate) enum IdKind {
+    /// An ID variant that is compatible with [`crate::entity::Entity`].
+    Entity = 0,
+    /// A future ID variant for Relationship Pairs.
+    Pair = 0b1000_0000,
```

Comment:
This should probably be cut from this PR before shipping it.

Reply:
To properly test the masking, this kinda needs to stay, but perhaps I could instead slap a `#[cfg(test)]` on it for now, so it is not exposed outside of tests? I could rename it to `Placeholder` as well.

Reply:
Thinking a bit more about this, I think this kinda has to stay, as this enum represents the two states of the most significant bit in the `Identifier`. A `1` is just not currently used for anything other than testing, but because we return the enum as the result of reading that bit, we need some way to denote the `1` or `0` states. However, `IdKind` and `IdentifierMask` are not `pub` and should remain in the crate, so this currently does not get exposed to the public API.

I'm definitely keeping it as `Placeholder`, as that's all it is at the moment.

Reply:
Yeah, let's swap it to `Placeholder` and then it can stay.